### PR TITLE
Make `bazel run` works with minimal mode

### LINKED ISSUE #11920: UX: make --remote_download_output=minimal download the necessary files when invoking `bazel run`
### Description of the problem / feature request:

When using remote build exection, calling `bazel run sometarget --remote_download_output=minimal` doesn't make sense: Bazel will not dowload all the files required to run `sometarget`.

I propose changing the behaviour of `--remote_download_output=minimal` to download files needed to run an action that is called with `bazel run`.

### Feature requests: what underlying problem are you trying to solve with this feature?

I'm trying to improve the user experience of Bazel.

Not downloading the files needed to run a target passed to `bazel run` is never something that a user wants to do.

### Any other information, logs, or outputs that you want to share?

I think the behavior would still be consistent with the meaning of `--remote_download_output=minimal`. The documentation says:

> If set to 'minimal' doesn't download any remote build outputs to the local machine, except the ones required by local actions.

I believe it is fair to consider `sometarget` a local action when calling `bazel run sometarget`.

### PR DESCRIPTION
**Pull request by coeuvre**
Always use `ToplevelArtifactsDownloader` when building without the bytes.

It checks the combination of current command (e.g. `build`, `run`, etc.) and download mode (e.g. `toplevel`, `minimal`) to decide whether download outputs for the toplevel targets or not.

Also in the `RunCommand`, we wait for the background downloads before checking the local filesystem.

Fixes #11920.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java [hunk 1]
```diff
@@ -929,7 +929,14 @@ public void executorInit(CommandEnvironment env, BuildRequest request, ExecutorB
       remoteOutputService.setActionInputFetcher(actionInputFetcher);
       actionContextProvider.setActionInputFetcher(actionInputFetcher);
 
-      if (remoteOutputsMode.downloadToplevelOutputsOnly()) {
+      boolean useToplevelDownloader = remoteOutputsMode.downloadToplevelOutputsOnly();
+
+      // Download toplevel artifacts for `run` command.
+      if ("run".equals(env.getCommandName())) {
```
Comment (coeuvre) [inline src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java:935]: We might need to support other commands (e.g. `coverage`, `mobile-install`, etc.). Do you think whether it is better if we **always** use toplevel downloader but pass in the download mode, let the downloaders decide what to download in both minimal/toplevel mode?

**Review by tjgq — APPROVED** [review]

File: src/main/java/com/google/devtools/build/lib/runtime/commands/RunCommand.java [hunk 2]
```diff
@@ -306,6 +306,14 @@ public BlazeCommandResult exec(CommandEnvironment env, OptionsParsingResult opti
       return BlazeCommandResult.detailedExitCode(result.getDetailedExitCode());
     }
 
+    if (env.getOutputService() != null) {
```
[the remark this answers is not in the document]
Comment (coeuvre) [reply src/main/java/com/google/devtools/build/lib/runtime/commands/RunCommand.java:312]: Done.

File: src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java (hunk 1 above)
Comment (coeuvre) [reply src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java]: Done.

**Comment by coeuvre** [comment]
@bazel-io flag

**Pull request closed without merging**